packages/backend/src/open_payments/payment/outgoing/service.ts

```diff
@@ -159,6 +178,138 @@ async function createOutgoingPayment(
   }
 }
 
+function validateAccessLimits(
+  payment: OutgoingPayment,
+  limits: AccessLimits
+): PaymentLimits | undefined {
+  if (
+    (!limits.receiver || payment.receiver === limits?.receiver) &&
+    validateAmountAssets(payment, limits)
+  ) {
+    let paymentInterval: Interval | undefined
+    if (limits.interval)
+      paymentInterval = getInterval(limits.interval, payment.createdAt)
+    if (!limits.interval || !!paymentInterval) {
+      return { ...limits, paymentInterval }
+    }
+  }
+}
+
+function validatePaymentInterval({
+  limits,
+  payment
+}: {
+  limits: PaymentLimits
+  payment: OutgoingPayment
+}): boolean {
+  return (
+    !limits.paymentInterval ||
+    (limits.paymentInterval.start.toMillis() <= payment.createdAt.getTime() &&
+      payment.createdAt.getTime() < limits.paymentInterval.end.toMillis())
+  )
+}
+
+function validateAmountAssets(
+  payment: OutgoingPayment,
+  limits: AccessLimits
+): boolean {
+  if (
+    limits.sendAmount &&
+    (limits.sendAmount.assetCode !== payment.asset.code ||
+      limits.sendAmount.assetScale !== payment.asset.scale)
+  ) {
+    return false
+  }
+  return (
+    (!limits.sendAmount && !limits.receiveAmount) ||
```

Comment:
I think this should be moved into `validateGrant` after the call to `validateAmountAssets` (to return true before querying other grant payments).